Wrappers/Python/test/test_functions.py

```diff
@@ -993,6 +995,35 @@ def test_L1Norm_vs_WeightedL1Norm(self):
         np.testing.assert_allclose(f1(x), float(N*M))
         np.testing.assert_allclose(f2(x), w)
         np.testing.assert_allclose(f2(x), f1(weights))
+        
+        np.random.seed(1)
+        weights= geom.allocate('random')
```

Comment:
Random weights could be negative. Consider changing this to `weights = geom.allocate('random'.abs()`

This also happens on lines 989 and 1032